airflow/hooks/hive_hooks.py

```diff
@@ -89,13 +91,58 @@ def run_cli(self, hql, schema=None):
                 stdout = ''
                 for line in iter(sp.stdout.readline, ''):
                     stdout += line
-                    logging.info(line.strip())
+                    if not test:
+                        logging.info(line.strip())
                 sp.wait()
 
                 if sp.returncode:
-                    raise AirflowException(all_err)
+                    if not test:
+                        raise AirflowException(all_err)
+                    else:
+                        return (False, stdout)
```

Comment:
It's confusing to have functions return tuples, and even more so to have them return heterogenous types depending on the call. If you run the call in a `try:` block you can essentially infer the true/false here